Content.Shared/Ninja/Systems/NinjaGlovesSystem.cs

```diff
@@ -0,0 +1,325 @@
+using Content.Shared.Damage.Components;
+using Content.Shared.Database;
+using Content.Shared.Doors.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.Electrocution;
+using Content.Shared.Emag.Systems;
+using Content.Shared.Examine;
+using Content.Shared.Hands.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+using Content.Shared.Tag;
+using Content.Shared.Timing;
+using Content.Shared.Toggleable;
+using Robust.Shared.Network;
+using Robust.Shared.Timing;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Content.Shared.Ninja.Systems;
+
+public abstract class SharedNinjaGlovesSystem : EntitySystem
+{
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly SharedCombatModeSystem _combatMode = default!;
+    [Dependency] protected readonly SharedDoAfterSystem _doafter = default!;
+    [Dependency] private readonly SharedElectrocutionSystem _electrocution = default!;
+    [Dependency] private readonly EmagSystem _emag = default!;
+    [Dependency] private readonly SharedInteractionSystem _interaction = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly SharedNinjaSystem _ninja = default!;
+    [Dependency] private readonly SharedPopupSystem _popups = default!;
+    [Dependency] private readonly TagSystem _tags = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NinjaGlovesComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<NinjaGlovesComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<NinjaGlovesComponent, ToggleActionEvent>(OnToggleAction);
+        SubscribeLocalEvent<NinjaGlovesComponent, GotUnequippedEvent>(OnUnequipped);
+        SubscribeLocalEvent<NinjaGlovesComponent, DoAfterEvent>(EndedDoAfter);
+
+        SubscribeLocalEvent<NinjaDoorjackComponent, InteractionAttemptEvent>(OnDoorjack);
+
+        SubscribeLocalEvent<NinjaStunComponent, InteractionAttemptEvent>(OnStun);
+
+        SubscribeLocalEvent<NinjaDrainComponent, InteractionAttemptEvent>(OnDrain);
+        SubscribeLocalEvent<NinjaDrainComponent, DoAfterEvent>(OnDrainDoAfter);
+
+        SubscribeLocalEvent<NinjaDownloadComponent, InteractionAttemptEvent>(OnDownload);
+        SubscribeLocalEvent<NinjaDownloadComponent, DoAfterEvent>(OnDownloadDoAfter);
+
+        SubscribeLocalEvent<NinjaTerrorComponent, InteractionAttemptEvent>(OnTerror);
+        SubscribeLocalEvent<NinjaTerrorComponent, DoAfterEvent>(OnTerrorDoAfter);
```

Comment:
```suggestion
        SubscribeLocalEvent<NinjaGlovesComponent, InteractionAttemptEvent>(OnInteractionAttempt)
        SubscribeLocalEvent<NinjaGlovesComponent, DoAfterEvent>(EndedDoAfter);

        SubscribeLocalEvent<NinjaDoorjackComponent, InteractionAttemptEvent>(OnDoorjack);

        SubscribeLocalEvent<NinjaStunComponent, InteractionAttemptEvent>(OnStun);

        SubscribeLocalEvent<NinjaDrainComponent, InteractionAttemptEvent>(OnDrain);
        SubscribeLocalEvent<NinjaDrainComponent, DoAfterEvent<DrainData>>(OnDrainDoAfter);

        SubscribeLocalEvent<NinjaDownloadComponent, InteractionAttemptEvent>(OnDownload);
        SubscribeLocalEvent<NinjaDownloadComponent, DoAfterEvent<DownloadData>>(OnDownloadDoAfter);

        SubscribeLocalEvent<NinjaTerrorComponent, InteractionAttemptEvent>(OnTerror);
        SubscribeLocalEvent<NinjaTerrorComponent, DoAfterEvent>(OnTerrorDoAfter);
```

With the way the DoAfterEvent works, you need to add generics to each of these since they're all on the same entity. If you have the DoAfterEvent as it is now, it'll raise all of the events no matter what you do. I gave some examples on how it would look like. Doesn't necessarily need to be called XData but that's just an example.

With that, they'd need record structs, even if they're empty. Check out some of the other doafters that use generics to get an idea. Forensic Pad should be one.

You can consolidate the InteractionAttempts by making try methods instead of separate components. If they need that much dependency with the gloves component they really shouldn't be separate. Examples below.